lib/logproto/logproto-proxied-text-server.c

```diff
@@ -189,31 +189,100 @@ _log_proto_proxied_text_server_add_aux_data(LogProtoProxiedTextServer *self, Log
   return;
 }
 
+static LogProtoPrepareAction
+log_proto_proxied_text_server_prepare(LogProtoServer *s, GIOCondition *cond, gint *timeout)
+{
+  LogProtoProxiedTextServer *self = (LogProtoProxiedTextServer *) s;
+
+  *cond = s->transport->cond;
+
+  if(self->handshake_done)
+    return log_proto_text_server_prepare_method(s, cond, timeout);
+
+  /* if there's no pending I/O in the transport layer, then we want to do a read */
+  if (*cond == 0)
+    *cond = G_IO_IN;
+
+  return LPPA_POLL_IO;
+}
+
+static inline LogProtoStatus
+_fetch_into_proxy_buffer(LogProtoProxiedTextServer *self)
+{
+  while(self->proxy_header_buff_len < PROXY_PROTO_HDR_MAX_LEN)
+    {
+      gssize rc = log_transport_read(self->super.super.super.transport,
+                                     &(self->proxy_header_buff[self->proxy_header_buff_len]),
+                                     sizeof(gchar), NULL);
+      if(rc < 0)
+        {
+          if (errno == EAGAIN)
+            return LPS_AGAIN;
+          else
+            {
+              msg_error("I/O error occurred while reading proxy header", evt_tag_int(EVT_TAG_FD,
+                        self->super.super.super.transport->fd),
+                        evt_tag_error(EVT_TAG_OSERROR));
+              return LPS_ERROR;
+            }
+        }
+
+      /* permissive termination */
+      if(rc == 0)
+        return LPS_SUCCESS;
+
+      self->proxy_header_buff_len++;
+      self->proxy_header_buff[self->proxy_header_buff_len] = '\0';
+      if (self->proxy_header_buff[self->proxy_header_buff_len - 1] == '\n')
+        {
+          return LPS_SUCCESS;
+        }
+
+    }
+
+  msg_error("PROXY proto header with invalid header length",
+            evt_tag_int("max_parsable_length", PROXY_PROTO_HDR_MAX_LEN),
+            evt_tag_int("max_length_by_spec", PROXY_PROTO_HDR_MAX_LEN_RFC),
+            evt_tag_long("length", self->proxy_header_buff_len),
+            evt_tag_str("header", (const gchar *)self->proxy_header_buff));
+  return LPS_ERROR;
+}
+
+static void
+_log_proto_proxied_text_server_switch_to_tls(LogProtoProxiedTextServer *self)
+{
+  if (!multitransport_switch((MultiTransport *)self->super.super.super.transport, transport_factory_tls_id()))
+    {
+      msg_error("proxied-tls failed to switch to TLS");
+      return;
+    }
+  msg_info("proxied-tls switch to TLS: OK");
+}
```

Comment:
Please move this to debug level (it was there previously, but it got reverted accidentally).